{{cookiecutter.project_slug}}/_/ci-services/deploy-stage/bitbucket-pipelines.yml

```diff
@@ -1,23 +1,105 @@
+  - script:
+    - &generate-secrets-vars
+      DJANGO_SECRET_KEY=$(cat /dev/urandom | tr -dc A-Za-z0-9 | head -c50)
+      DATABASE_PASSWORD=$(cat /dev/urandom | tr -dc A-Za-z0-9 | head -c16)
+      APPLICATION=${APPLICATION:-application{% if cookiecutter.environment_strategy == 'shared' %}-${BITBUCKET_DEPLOYMENT_ENVIRONMENT}{% endif %}}
+      DATABASE_HOST=${DATABASE_HOST:-{{ cookiecutter.database|lower }}{% if cookiecutter.environment_strategy == 'shared' %}-${BITBUCKET_DEPLOYMENT_ENVIRONMENT}{% endif %}}
+      DATABASE_NAME={{ cookiecutter.project_slug.replace('-', '_') }}
+      DATABASE_USER={{ cookiecutter.project_slug.replace('-', '_') }}
+    - &generate-secrets-app
+      oc -n ${TARGET} get secret ${APPLICATION} ||
+      oc -n ${TARGET} create secret generic ${APPLICATION}
+        --from-literal=DJANGO_DATABASE_URL={{ cookiecutter.database|lower }}://${DATABASE_USER}:${DATABASE_PASSWORD}@${DATABASE_HOST}/${DATABASE_NAME}
+        --from-literal=DJANGO_SECRET_KEY=${DJANGO_SECRET_KEY}
+  {%- if cookiecutter.monitoring == 'Sentry' %}
+        --from-literal=SENTRY_DSN=${SENTRY_DSN}
+  {%- endif %}
+    - &generate-secrets-db
+      oc -n ${TARGET} get secret ${DATABASE_HOST} ||
+      oc -n ${TARGET} create secret generic ${DATABASE_HOST}
+        --from-literal=POSTGRESQL_DATABASE=${DATABASE_NAME}
+        --from-literal=POSTGRESQL_USERNAME=${DATABASE_USER}
+        --from-literal=POSTGRESQL_PASSWORD=${DATABASE_PASSWORD}
 
-  branches:
-    master:
-    - parallel: *checks
-    - parallel: *tests
+  - step: &deploy-review-app
+      name: Deploy Review App
```

Comment:
Where/how is the review app deleted again?

Reply:
You can see it for the output of the deployment step. There are links to click on.

Care to find out yourself? :smiley_cat: :crossed_fingers:

Reply:
Sorry, misread your question. There are instructions in the deployment steps. No automation possible. We could implement a "custom" action.

Reply:
Ah, I see. Well too bad.